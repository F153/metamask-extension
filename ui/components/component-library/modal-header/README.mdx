import { Story, Canvas, ArgsTable } from '@storybook/addon-docs';

import { Text } from '..';

import { ModalHeader } from './modal-header';

# ModalHeader

The `ModalHeader` is used as feedback text under a form field including error, success, warning or info messages

<Canvas>
  <Story id="components-componentlibrary-modalheader--default-story" />
</Canvas>

## Props

The `ModalHeader` accepts all props below as well as all [Box](/docs/components-ui-box--default-story#props) component props

<ArgsTable of={ModalHeader} />

`ModalHeader` accepts all [Text](/docs/components-componentlibrary-text--default-story#props) component props

<ArgsTable of={Text} />

### Children

Use the `children` prop to render the content of the `ModalHeader`. If the type of `children`
