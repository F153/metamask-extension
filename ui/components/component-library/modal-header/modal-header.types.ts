import { HeaderBaseProps } from '../header-base';

export interface ModalHeaderProps extends HeaderBaseProps {
  /**
   * The onClose event handler for the ModalHeader
   */
  onClickCloseButton?: () => void;
  /**
   * The props to apply to the close button ButtonIcon
   */
  closeButtonProps?: any; // TODO: Update to ButtonIconProps
  /**
   * The onBack event handler for the ModalHeader
   */
  onClickBackButton?: () => void;
  /**
   * The props to apply to the back button ButtonIcon
   */
  backButtonProps?: any; // TODO: Update to ButtonIconProps
  /**
   * Replaces the back button with a custom element
   */
  startAccessory?: React.ReactNode;
  /**
   * Replaces the close button with a custom element
   */
  endAccessory?: React.ReactNode;
  /**
   * The children is the title area of the ModalHeader
   */
  children?: React.ReactNode;
  /**
   * An additional className to apply to the ModalHeader
   */
  className?: string;
}
