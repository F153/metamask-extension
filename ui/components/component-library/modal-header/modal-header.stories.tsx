import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';

import { Text } from '..';
import { ModalHeader } from './modal-header';

import README from './README.mdx';

export default {
  title: 'Components/ComponentLibrary/ModalHeader',
  component: ModalHeader,
  parameters: {
    docs: {
      page: README,
    },
  },
  argTypes: {
    children: {
      control: 'text',
    },
    className: {
      control: 'text',
    },
    onClickCloseButton: {
      action: 'onClickCloseButton',
    },
    onClickBackButton: {
      action: 'onClickBackButton',
    },
  },
  args: {
    children: 'Modal Header',
  },
} as ComponentMeta<typeof ModalHeader>;

const Template: ComponentStory<typeof ModalHeader> = (args) => (
  <ModalHeader {...args} />
);

export const DefaultStory = Template.bind({});
DefaultStory.storyName = 'Default';

export const Children = Template.bind({});
Children.args = {
  onClickBackButton: null,
  onClickCloseButton: null,
};

export const onClickCloseButton = Template.bind({});
onClickCloseButton.args = {
  onClickBackButton: null,
};

export const onClickBackButton = Template.bind({});
onClickBackButton.args = {
  onClickCloseButton: null,
};

export const StartAccessoryAndEndAccessory = Template.bind({});
StartAccessoryAndEndAccessory.args = {
  startAccessory: <Text>Start Accessory</Text>,
  endAccessory: <Text>End Accessory</Text>,
};
