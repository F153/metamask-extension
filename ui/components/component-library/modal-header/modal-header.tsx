import React from 'react';
import classnames from 'classnames';

import { HeaderBase, ButtonIcon, BUTTON_ICON_SIZES, IconName } from '..';

import { ModalHeaderProps } from './modal-header.types';

export const ModalHeader: React.FC<ModalHeaderProps> = ({
  className = '',
  children,
  onClickCloseButton,
  closeButtonProps,
  onClickBackButton,
  backButtonProps,
  startAccessory,
  endAccessory,
  ...props
}) => {
  return (
    <HeaderBase
      className={classnames('mm-modal-header', className)}
      startAccessory={
        startAccessory ||
        (onClickBackButton && (
          <ButtonIcon
            iconName={IconName.ArrowLeft}
            onClick={onClickBackButton}
            size={BUTTON_ICON_SIZES.SM}
            {...backButtonProps}
          />
        ))
      }
      endAccessory={
        endAccessory ||
        (onClickCloseButton && (
          <ButtonIcon
            iconName={IconName.Close}
            onClick={onClickCloseButton}
            size={BUTTON_ICON_SIZES.SM}
            {...closeButtonProps}
          />
        ))
      }
      {...props}
    >
      {children}
    </HeaderBase>
  );
};
