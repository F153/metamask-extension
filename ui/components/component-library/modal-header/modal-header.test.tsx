/* eslint-disable jest/require-top-level-describe */
import { render } from '@testing-library/react';
import React from 'react';

import { ModalHeader } from './modal-header';

describe('ModalHeader', () => {
  it('should render with text inside the ModalHeader', () => {
    const { getByText, getByTestId } = render(
      <ModalHeader data-testid="test">modal header</ModalHeader>,
    );
    expect(getByText('modal header')).toBeDefined();
    expect(getByTestId('test')).toHaveClass('mm-modal-header');
  });
  it('should match snapshot', () => {
    const { container } = render(<ModalHeader>modal header</ModalHeader>);
    expect(container).toMatchSnapshot();
  });
  it('should render with and additional className', () => {
    const { getByTestId } = render(
      <ModalHeader data-testid="test" className="test">
        modal header
      </ModalHeader>,
    );
    expect(getByTestId('test')).toHaveClass('test');
  });
  it('should render with react nodes inside the ModalHeader', () => {
    const { getByText } = render(
      <ModalHeader>
        <span>modal header</span>
      </ModalHeader>,
    );
    expect(getByText('modal header')).toBeDefined();
  });
});
