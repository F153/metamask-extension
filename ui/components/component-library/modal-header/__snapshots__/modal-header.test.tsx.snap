// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalHeader should match snapshot 1`] = `
<div>
  <div
    class="box mm-header-base mm-modal-header box--display-flex box--flex-direction-row box--justify-content-space-between"
  >
    <h4
      class="box mm-text mm-text--heading-sm mm-text--text-align-center box--flex-direction-row box--width-full box--color-text-default"
    >
      modal header
    </h4>
  </div>
</div>
`;
